packages/meditrak-app-server/src/__tests__/utilities/setupTestApp.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { TestableServer } from '@tupaia/server-boilerplate';
+import { getTestDatabase } from '@tupaia/database';
+
+import { createApp } from '../../app';
+import { TEST_DATA } from '../__integration__/fixtures';
+import { AuthApiMock } from './AuthApiMock';
+import { CentralApiMock } from './CentralApiMock';
+
+jest.mock('@tupaia/api-client', () => {
+  const original = jest.requireActual('@tupaia/api-client'); // Step 2.
+  return {
+    ...original,
+    TupaiaApiClient: jest.fn().mockImplementation(() => {
+      return {
+        auth: new AuthApiMock(TEST_DATA),
+        central: new CentralApiMock(),
```

Comment:
Decided to go with some really simple mock classes here. I find mocking to be a pain in terms of code design honestly. Either you:
a) try and keep your mocks close to the source, which involves exposing functionality and increased coupling
b) try and keep your mocks convenient for your tests, which makes them less 'true to life'

Personally, in this case (and perhaps in general) I prefer to mock as little as possible, but to go with option b) and allow them to drift from the source. So I guess I'd anticipate each package might have their own version of mocks for each `ApiConnection` they use.

Reply:
I'm with you here on having pretty lightweight mocking, and as a result getting more code under test (even if you lose some test specificity)

Reply:
When the mocks become as complicated and tricky to maintain the the code they're meant to stub, then I always feel like I've missed the point. Don't wanna end up in that position 👍